src/gluonnlp/attention_cell.py

```diff
@@ -670,6 +670,161 @@ def __repr__(self):
                         dtype=self._dtype)
 
 
+
+
+def multi_head_sliding_window_dot_attn(F, query, key, value, dilation, valid_length,
+                                       w: int, symmetric: bool = True,
+                                       dropout: float = 0.0, scaled: bool = True,
+                                       normalized: bool = False, eps: float = 1E-6,
+                                       query_head_units: Optional[int] = None,
+                                       layout: str = 'NKT',
+                                       dtype=np.float32):
+    """Multihead sliding window attention between the query, key and value,
+    described at *Longformer: The Long-Document Transformer*,
+    available at https://arxiv.org/pdf/2004.05150.pdf.
+
+    Given a fixed window size *2w*, each token attends to *w* tokens on the left side
+    if using causal attention (setting *symmetric* to *False*),
+    otherwise each token attends to *w* tokens on each side.
+
+    Parameters
+    ----------
+    F
+    query
+        Query. The shape is (batch_size, seq_length, num_heads, num_head_units)
+
+    key
+        Key. The shape is (batch_size, seq_length, num_heads, num_head_units)
+    value
+        Value. The shape is (batch_size, seq_length, num_heads, num_head_units)
+    dilation
+        Dilation. The shape is (num_heads,)
+    valid_length
+        Valid length. The shape is (batch_size,)
+    w
+        The one-sided window length.
+    symmetric
```

Comment:
How about to call it `causal`?

Reply:
why causal?

Reply:
Causal attention means that the input itself won't be used. symmetric is also fine.

Reply:
either one-sided or double sided uses itself.

Reply:
Yeah, then we should mention that it won't use any tokens in the future in the documentation.